core/domain/topic_domain.py

```diff
@@ -155,16 +155,93 @@ def to_dict(self):
             'version': self.version
         }
 
+    @classmethod
+    def require_valid_topic_id(cls, topic_id):
+        """Checks whether the topic id is a valid one.
+
+        Args:
+            topic_id: str. The topic id to validate.
+        """
+        if not isinstance(topic_id, basestring):
+            raise utils.ValidationError(
+                'Topic id should be a string, received: %s' % topic_id)
+
+        if len(topic_id) != 12:
+            raise utils.ValidationError('Topic id %s is invalid' % topic_id)
+
+    @classmethod
+    def require_valid_name(cls, name):
+        """Checks whether the name of the topic is a valid one.
+
+        Args:
+            name: str. The name to validate.
+        """
+        if not isinstance(name, basestring):
+            raise utils.ValidationError('Name should be a string.')
+
+        if name == '':
+            raise utils.ValidationError('Name field should not be empty')
+
+    def delete_skill(self, skill_id):
+        """Removes a skill from the skill_ids list.
+
+        Args:
+            skill_id: str. The skill id to remove from the list.
+
+        Returns:
+            list. The updated list of skill ids.
+        """
+        skill_ids = self.skill_ids
+        skill_ids.remove(skill_id)
+        return skill_ids
+
+    def add_skill(self, skill_id):
+        """Adds a skill to the skill_ids list.
+
+        Args:
+            skill_id: str. The skill id to add to the list.
+
+        Returns:
+            list. The updated list of skill ids.
+        """
+        skill_ids = self.skill_ids
+        skill_ids.append(skill_id)
+        return skill_ids
```

Comment:
Ditto, why return? Also might want to check if skill_id is there in the first place and raise exception if so.